zkvm/docs/zkvm-spec.md

```diff
@@ -828,6 +787,39 @@ V == v·B + 0·B2
 1. Prover shows `v`.
 2. Verifier checks equality `V == v·B`.
 
+### Taproot
+Based on Gregory Maxwell's [Taproot proposal](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2018-January/015614.html), Taproot provides efficient and privacy-preserving storage of smart contracts. Multi-party blockchain contracts typically have a top-level "success clause" (all parties agree and sign) or "alternative clauses" to let a party exit the contract based on pre-determined constraints. Taproot has three significant features.
+
+1. The alternative clauses `{C_i}`, each of which is a [program](#program), are stored as [blinded programs](#blinded-program) and compressed in a [Merkle tree](#merkle-binary-tree) with root `M`.
+2. A signing key `X` and the Merkle root `M` (from 1) are committed to a single signing key `P` using a hash function `h1`, such that `P = X + h1(X, M)`. This makes signing for `P` possible if parties want to sign for `X` and avoids revealing the alternative clauses.
+3. Calling a program will check a [call proof](#call-proof) to verify the program's inclusion in the Taproot tree before executing the program.
+
+The commitment scheme is defined using the [transcript](#transcript) protocol by committing (1) the signing key, as a compressed 32-byte point, and (2) the root of the Merkle tree constructed from the `n` programs and (3) squeezing a scalar bound to all `n` programs.
+
+```
+// given Merkle root M and signing key X
+t = Transcript("ZkVM.taproot")
+t.commit_bytes(b"key", X; // Compressed Ristretto
+t.commit_bytes(b"merkle", M); // [u8; 32]
+t.challenge_scalar(b"h")
+```
+
+In this context, navigating to a branch involves selecting a leaf node program to run. A branch is chosen using the [`call`](#call) instruction, which also takes a program and [call proof](#call-proof). If, using this information, the relation `P = X + h1(X, M)` is verified, then the associated program will be executed.
+
+
+### Call proof
+An argument to the [`call`](#call) instruction, this proof is an extension of the Merkle path and contains the information to verify that a program is represented in the [Taproot](#taproot) Merkle tree. It contains `X`, the signing key; a list of left-right neighbors, each a LE32 item, but excluding the program leaf and root hash; and a LE32 bit pattern, indicating each neighbor's position.
+
+### Blinded program
+The blinded program is used for efficient construction of the Merkle tree, and it can either contain a program or its accompanying blinding factor. Thus, blinding factors are simply even items in the Merkle tree. We add the computation of the per-clause blinding factor, and we commit them as follows.
```

Comment:
Each clause in this paragraph feels like it needs to be its own paragraph. How is the blinded program used for efficient construction of the Merkle tree? How does it "contain" things? How does "blinding factors are simply even items" follow from that?